hepcrawl/parsers/crossref.py

```diff
@@ -0,0 +1,387 @@
+        else:
+            return None
+
+    @property
+    def number_of_pages(self):
+        first_page = self.page_start
+        last_page = self.page_end
+
+        if first_page and last_page:
+            return int(self.page_end) - int(self.page_start) or 1
+
+    @staticmethod
+    def get_author_name(author_node):
+        """Extract an author's name."""
+        author_name = ', '.join([author_node.get("family"),author_node.get("given")])
+
+        return author_name
+
+    def get_author_affiliations(self, author_node):
+        """Extract an author's affiliations."""
+        affiliations = force_list(author_node.get("affiliation"))
+
+        auth_aff = []
+        for affiliation in affiliations:
+            auth_aff.append(affiliation.get('name'))
+
+        return auth_aff
+
+
+    def get_author_uuid(self, author_node):
+        """Extract an author's uuid."""
+        orcid_value = author_node.get('ORCID')
+
+        if orcid_value:
+            return [('ORCID',author_node.get("ORCID"))]
+        else:
+            return None
+
+    def get_author(self, author_node):
+        """Extract one author.
+
+        Args:
+            author_node(dict): a dictionary on a single author.
+
+        Returns:
+            dict: the parsed author, conforming to the Inspire schema.
+        """
+        author_name = self.get_author_name(author_node)
+        affiliations = self.get_author_affiliations(author_node)
+        uuid = self.get_author_uuid(author_node)
+
+        return self.builder.make_author(author_name, raw_affiliations=affiliations, ids=uuid)
+
+    @property
+    def authors(self):
+        author_nodes = self.record.get("author")
+        authors = [self.get_author(author) for author in force_list(author_nodes)]
+
+        return authors
+
+    @property
+    def license(self):
+        license_nodes = self.record.get("license")
+        licenses = [self.get_license(license) for license in force_list(license_nodes)]
+
+        return licenses
+
+    def get_license(self, license_node):
+        """Extract one license.
+
+        Args:
+            license_node(dict): a dictionary on a single license.
+
+        Returns:
+            dict: the parsed license, conforming to the Inspire schema.
+        """
+        license = {
+            'imposing': self.publisher,
+            'material': self.material,
+            'url': self.get_license_url(license_node),
+        }
+
+        return license
+
+    def get_license_url(seld, license_node):
+        return license_node.get("URL")
+
+    @property
+    def publisher(self):
+        return self.record.get("publisher")
+
+    @property
+    def abstract(self):
+        return self.record.get("abstract")
+
+    @property
+    def imprints(self):
+        date_parts = self.record.get("issued")
+
+        if date_parts:
+            date = date_parts.get("date-parts")[0]
+            if None not in date:
+                return '-'.join(map(str,date))
```

Comment:
this is not how you should do it. Have a look at `PartialDate` from `inspire-utils.date`.